src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -0,0 +1,138 @@
+
+            $nodeFinder->find($ast, function (Node $node) use(&$extractedArguments): bool
+            {
+                if (
+                    ($node instanceof Node\Expr\New_ && false !== \array_search('TranslatableMessage', $node->class->parts))
+                    || ($node instanceof Node\Expr\MethodCall && $node->name->toString() === 'trans')
+                    || ($node instanceof Node\Expr\FuncCall && $node->name->toString() === 't')
+                ) {
+                    $extractedArguments[] = $this->extractFromArgs($node);
+                }
+
+                return true;
+            });
+
+            foreach ($extractedArguments as $args) {
+                $catalogue->set($args['message'], $this->prefix.$args['message'], $args['domain']);
+                $metadata = $catalogue->getMetadata($args['message'], $args['domain']) ?? [];
+                $normalizedFilename = preg_replace('{[\\\\/]+}', '/', $file);
+                $metadata['sources'][] = $normalizedFilename.':'.$args['lineNumber'];
+                $catalogue->setMetadata($args['message'], $metadata, $args['domain']);
+            }
+
+            gc_mem_caches();
+        }
+    }
+
+    public function setPrefix(string $prefix)
+    {
+        $this->prefix = $prefix;
+    }
+
+    protected function canBeExtracted(string $file)
+    {
+        return $this->isFile($file) && 'php' === pathinfo($file, \PATHINFO_EXTENSION);
+    }
+
+    protected function extractFromDirectory(array|string $resource)
+    {
+        if (!class_exists(Finder::class)) {
+            throw new \LogicException(sprintf('You cannot use "%s" as the "symfony/finder" package is not installed. Try running "composer require symfony/finder".', static::class));
+        }
+
+        $finder = new Finder();
+
+        return $finder->files()->name('*.php')->in($resource);
+    }
+
+    private function extractFromArgs(Node\Expr\New_|Node\Expr\MethodCall|Node\Expr\FuncCall $node): array
+    {
+        foreach ($args = $node->getRawArgs() as $arg) {
+            switch ($arg->name) {
+                case 'message':
+                    $message = $this->extractValue($arg);
+                    break;
+                case 'domain':
+                    $domain = $this->extractValue($arg);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        $message = $message ?? $this->extractValue($args[0]);
+        $domain = $domain ?? (isset($args[2]) ? $this->extractValue($args[2]) : null) ?? 'messages';
+
+        return [
+            'message' => $message,
+            'domain' => $domain,
+            'lineNumber' => $node->getAttribute('startLine'),
+        ];
+    }
+
+    private function extractValue(Node\Arg $arg): ?string
+    {
+        $result = '';
+
+        if ($arg->value instanceof Node\Expr\BinaryOp\Concat) {
+            $left = $arg->value;
+            while (!$left instanceof Node\Scalar\String_) {
```

Comment:
this looks weird to me. It seems to assume that the left of a Concat operator can either be a string literal or another Concat operator, which is clearly wrong.